### PR TITLE
[ML][Pipelines] Refine some CLI error message

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

Before:
image
image

After:
image

image

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel_for.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def __init__(
             self._outputs = self._build_outputs_dict_without_meta(outputs=actual_outputs)
 
         self._items = items
-        self._validate_items(raise_error=True)
+        # self._validate_items(raise_error=True)
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel_for.py:85]: so we'll put off items validation? then why keep this line?

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/pipeline/pipeline_component.py [hunk 2]
```diff
@@ -107,21 +107,18 @@ def _post_load_pipeline_jobs(context, data: dict) -> dict:
...
                 # Convert to do-while node.
                 job_instance = DoWhile._create_instance_from_schema_dict(pipeline_jobs=jobs, loaded_data=job_instance)
-                jobs[key] = job_instance
             elif job_instance.get("type") == ControlFlowType.PARALLEL_FOR:
                 # Convert to do-while node.
                 job_instance = ParallelFor._create_instance_from_schema_dict(
                     pipeline_jobs=jobs, loaded_data=job_instance
                 )
-                jobs[key] = job_instance
+            jobs[key] = job_instance
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/pipeline/pipeline_component.py:121]: why this change can refine the CLI error message?

**Comment by wangchao1230** [comment]
Please take a look at the CI failure

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel_for.py (hunk 1 above)
Comment (D-W-) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel_for.py:85]: Good catch, forgot to remove this..

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/pipeline/pipeline_component.py (hunk 2 above)
Comment (D-W-) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/pipeline/pipeline_component.py:121]: This is not related to error message, just to fix your previous comment. Forgot to mention you here.
